tests/Writer/CategoriesTest.php

```diff
@@ -19,22 +19,46 @@ public function test_categories_xml(): void
         $xml->startDocument();
         $xml->startCatalog('TestCatalog');
 
-        foreach (['Socks', 'Death Stars', 'Donuts'] as $index => $example) {
-            $element = new Category('CAT' . $index);
-            $element->setDisplayName($example);
+        $categories = [
+            'Socks' => [
+                'index' => 0,
+                'dates' => [
+                    new DateTimeImmutable('2018-01-01 01:01:01'),
+                    new DateTimeImmutable('2018-02-02 02:02:02'),
+                ],
+            ],
+            'Death Stars' => [
+                'index' => 1,
+                'dates' => [
+                    null,
+                    new DateTimeImmutable('2018-02-02 02:02:02'),
+                ],
+            ],
+            'Donuts' => [
+                'index' => 2,
+                'dates' => [
+                    new DateTimeImmutable('2018-01-01 01:01:01'),
+                    null,
```

Comment:
as it's just nilIfEmpty rather than nilIfNull, is it worth an empty string test?

Reply:
Added some extra tests